cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -133,7 +133,20 @@ private void silentDisconnect() {
         }
     }
 
+    /**
+     * Reset the current session. This rolls back any open transactions.
+     */
     public void reset() {
-//        session.reset();
+        // TODO once drivers release next milestone
+        //if (session != null) {
+            // session.reset();
+        //}
+
+        // Clear current state
```

Comment:
I do too. The bolt docs are not 100% clear. It is fine to mark a failed transaction as failed an infinite time at least.